docs/upgrade-guide-1.6-to-1.7.ipynb

```diff
@@ -117,7 +153,117 @@
    },
    "source": [
     "## Upgrade Istio\n",
-    "TBD"
+    "\n",
+    "Upgrade of Istio service mesh components is performed according to Istio instructions and invloves upgrading charms in sequence. Note that `istio-gateway` charm should always me removed before starting upgrade. For more details on Istio upgrade and how to debug failed upgrades refer to [this document](https://github.com/canonical/istio-operators/blob/main/charms/istio-pilot/README.md). It is assumed that `istio-pilot` version in Charmed Kubeflow deployed is 1.11.\n",
+    "\n",
+    "1. Remove `istio-ingressgaeway` application and corresponding relation with `istio-pilot`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "code"
+    ],
+    "vscode": {
+     "languageId": "plaintext"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# remove relation and istio-ingressgateway application\n",
+    "juju remove-relation istio-pilot istio-ingressgateway\n",
+    "juju remove-application istio-ingressgateway"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": [
+     "text"
+    ]
+   },
+   "source": [
+    "2. Ensure that `istio-ingressgateway` application and all related resources are properly removed. The following commands should succeed (return `0`):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "code"
+    ],
+    "vscode": {
+     "languageId": "plaintext"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "juju show-application istio-ingressgateway 2> >(grep -q \"not found\" && echo $?)\n",
+    "kubectl -n kubeflow get deploy istio-ingressgateway-workload 2> >(grep -q \"NotFound\" && echo $?)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": [
+     "text"
+    ]
+   },
+   "source": [
+    "*Troubleshooting of removal of `istio-ingressgateway` application*\n",
+    "\n",
+    "If required, remove `istio-ingressgateway` application with `--force` option and remove `istio-ingressgateway-workload` manually:"
```

Comment:
I'd expand this section, this should be the last resource if the application is not removed. I'd add at least a warning and possible issues when force removing an application.